src/components/form/FlexList.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import * as selectors from '@/reducers/selectors';
+import * as actions from '@/actions/creators';
+
+import FormFieldBuilder from './FormFieldBuilder';
+
+const mapStateToProps = (state, ownProps) => {
+  const selectedSection = selectors.getSelectedSectionId(state);
+  return {
+    canAddItem: selectors.getCanAddListItem(state, selectedSection, ownProps.listName),
+    canDeleteItem: selectors.getCanDeleteListItem(state, selectedSection, ownProps.listName),
+    nestedFields: ownProps.fieldStructure.fields.map(fieldName => {
+      return selectors.getFieldByName(state, fieldName);
+    }),
+    selectedSection,
+  };
+}
+
+const mapDispatchToProps = (dispatch) => ({
+  onAddItem: (sectionId, listName) => dispatch(actions.addListItem(sectionId, listName)),
+  onDeleteItem: (sectionId, listName, itemIndex) => dispatch(actions.deleteListItem(sectionId, listName, itemIndex)),
+});
+
+const FlexList = ({
+  canAddItem,
+  canDeleteItem,
+  fields: reduxFormFields,
+  listName,
+  fieldStructure,
+  nestedFields,
+  selectedSection,
+  onAddItem,
+  onDeleteItem,
+}) => (
+  <section className='form__list'>
+    <span className='form__label'>{fieldStructure.label}:</span>
+      {reduxFormFields.map((f, index) => {
+        return (
+          <article key={index} className='list-item'>
```

Comment:
We should never use `index` as a `key`. Do the `reduxFormFields` have any unique identifier or can we otherwise give them one?